### PR TITLE
Cole/BFB-151: Remade PR

### PR DESCRIPTION
**Pull request by csweet22**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-151

**Description**
Remaking this PR. Edits from past PR comments. Also added Random Seed to Height Map Shader offset.

### CODE REVIEW

**Comment by m4rtint** [comment]
Some of the scenes are broken because you haven't update the scripts yet, can you set them up in 
- WeaponRaceCourse
- GunRaenge
- MS2_RaceCourse
- WeaponSandboxSceneTemplate
image

File: Assets/_BForBoss/_VFX/Scripts/DemoPatrolBehaviour.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System.Collections;
+using System.Collections.Generic;
+using PerigonGames;
+using UnityEngine;
+using UnityEngine.Serialization;
+
+namespace BForBoss
```
Comment (m4rtint) [inline Assets/_BForBoss/_VFX/Scripts/DemoPatrolBehaviour.cs:13]: Can you rename this to Perigon.VFX

same with the other script

**Comment by m4rtint** [comment]
image
I see you added a wire vine builder? 
That woudl be a different ticket :P

File: Assets/_BForBoss/_Weapons/Scripts/Weapons/WeaponBehaviour.cs [hunk 2]
```diff
@@ -12,6 +13,7 @@ public abstract class WeaponBehaviour : MonoBehaviour
         private readonly Vector3 CenterOfCameraPosition = new Vector3(0.5f, 0.5f, 0);
         [SerializeField] protected Transform _firePoint = null;
         [SerializeField] private CrosshairBehaviour _crosshair = null;
+        [SerializeField] private VisualEffect _muzzleFlash = null;
```
Comment (m4rtint) [inline Assets/_BForBoss/_Weapons/Scripts/Weapons/WeaponBehaviour.cs:16]: Actually, can you move this into `WeaponScriptableObject` 

it's a shareable object, where if you update this object in one place, it get's applied everywhere else.

Comment (csweet22) [reply Assets/_BForBoss/_Weapons/Scripts/Weapons/WeaponBehaviour.cs:16]: That's actually what I originally had but I couldn't figure out how to get it to save the reference correctly. I can try again though.

**Comment by csweet22** [comment]
> 

Oh! The actual content isn't there. That is just because I pushed the parent folders metadata because I wasn't sure if the metadata and newly named DemoPatrolBehaviour would play nice together if I didn't update the metadata.

File: Assets/_BForBoss/_Weapons/Scripts/Weapons/WeaponBehaviour.cs (hunk 2 above)
Comment (m4rtint) [reply Assets/_BForBoss/_Weapons/Scripts/Weapons/WeaponBehaviour.cs:16]: After some back and forth, realized this is the best solution 👍

**Review by m4rtint — APPROVED** [review]
Nice lgtm :)

**Review by m4rtint — CHANGES_REQUESTED** [review]
Wait, i don't seem to be able to find the scripts you put in

**Review by m4rtint — APPROVED** [review]
nvm. Github

**Pull request merged**